src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -67,6 +67,15 @@ public virtual void AddCompilationRoots()
 
                 AddCompilationRootsForExports(module);
             }
+
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
+                if (_typeSystemContext.InputFilePaths.ContainsKey(inputFile.Value)) continue;
```

Comment:
Nope, the issue is because the same module shows up twice within the ReferencePaths.